tests/system/python/api/test_notification.py

```diff
@@ -0,0 +1,229 @@
+        ({"name": "Test4", "description": "Test4_Notification", "rule": NOTIFY_INBUILT_RULES[1], "enabled": True,
+          "notification_type": "one shot"}, '400: Missing channel property in payload.'),
+        ({"name": "Test4", "description": "Test4_Notification", "rule": NOTIFY_INBUILT_RULES[1],
+          "channel": NOTIFY_PLUGIN, "enabled": True}, '400: Missing notification_type property in payload.'),
+        ({"name": "=", "description": "Test4_Notification", "rule": NOTIFY_INBUILT_RULES[1], "channel": NOTIFY_PLUGIN,
+          "enabled": True, "notification_type": "one shot"}, '400: Invalid name property in payload.'),
+        ({"name": "Test4", "description": "Test4_Notification", "rule": "+", "channel": NOTIFY_PLUGIN, "enabled": True,
+          "notification_type": "one shot"}, '400: Invalid rule property in payload.'),
+        ({"name": "Test4", "description": "Test4_Notification", "rule": NOTIFY_INBUILT_RULES[1], "channel": ":",
+          "enabled": True, "notification_type": "one shot"}, '400: Invalid channel property in payload.'),
+        ({"name": "Test4", "description": "Test4_Notification", "rule": NOTIFY_INBUILT_RULES[1],
+          "channel": NOTIFY_PLUGIN, "enabled": "bla", "notification_type": "one shot"},
+         '400: Only "true", "false", true, false are allowed for value of enabled.'),
+        ({"name": "Test4", "description": "Test4_Notification", "rule": "InvalidRulePlugin",
+          "channel": "InvalidChannelPlugin", "enabled": True, "notification_type": "one shot"},
+         '400: Invalid rule plugin InvalidRulePlugin and/or delivery plugin InvalidChannelPlugin supplied.')
+    ])
+    def test_invalid_create_notification_instance(self, foglamp_url, test_input, expected_error):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("POST", '/foglamp/notification', json.dumps(test_input))
+        r = conn.getresponse()
+        assert 400 == r.status
+        r = r.read().decode()
+        assert expected_error == r
+
+    def test_create_valid_notification_instance(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("POST", '/foglamp/notification', json.dumps(data))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert "Notification {} created successfully".format(data['name']) == jdoc['result']
+
+        conn.request("GET", '/foglamp/notification')
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert data['name'] == jdoc['notifications'][0]['name']
+        assert data['channel'] == jdoc['notifications'][0]['channel']
+        assert 'true' == jdoc['notifications'][0]['enable']
+        assert data['notification_type'] == jdoc['notifications'][0]['notificationType']
+        assert data['rule'] == jdoc['notifications'][0]['rule']
```

Comment:
May be `actual = jdoc['notifications'][0]` to write less in all 6 assertions
assert data['name'] == actual["name"]